### PR TITLE
[M2032] change GoI weighting to equal split

### PR DESCRIPTION
**Pull request by saanobhaai**
## Summary by CodeRabbit

* **Bug Fixes**
  * Updated group-of-interest attribution for family, order, and class to use equal-split weighting across distinct matching names, improving consistency in reported densities.
  * Fixed an edge case where higher-level taxa could be omitted from group-of-interest density results while related abundance totals were still present.

* **Tests**
  * Expanded fixtures and revised expectations to cover equal-split scenarios and families with no genus GoI mapping.
  * Added a new test to verify correct handling when group-of-interest density should be absent.

### CODE REVIEW

**Comment by saanobhaai** [comment]
[change GoI density weighting](https://trello.com/c/ovp8OLA3/2032-change-goi-density-weighting)

File: src/api/models/sql_models/beltinvert.py [hunk 1]
```diff
@@ -183,6 +186,10 @@ class BeltInvertSUSQLModel(BaseSUSQLModel):
             JOIN invert_genus g ON o.invert_attribute_id = g.invertattribute_ptr_id
             JOIN invert_group_of_interest goi ON g.group_of_interest_id = goi.invertattribute_ptr_id
             UNION ALL
+            -- NOTE: If a family/order/class taxon has no genera with GoI assignments in the
+            -- taxonomy DB, the JOIN below produces no rows and the observation's count is
+            -- silently absent from density_indha_group_interest while still counted in
+            -- total_abundance and density_indha, making those fields inconsistent.
```
Comment (JBurkinshaw) [inline src/api/models/sql_models/beltinvert.py:192]: I like the addition of this note, but I don't think what it describes isn't tested, so it's currently just an worry in a comment. Could we add a fixture with a family-level obs whose genera have no GoI assignment, asserting the count stays in `total_abundance`/`density_indha` but is absent from `density_indha_group_interest`? That makes the current behaviour more intentional (unless you have a plan to change it in the future.

**Review by JBurkinshaw — APPROVED** [review]
